### PR TITLE
Update imdb data link and md5 hash

### PR DESCRIPTION
**Pull request by Bensk1**
I also updated the JSON files describing the schema of the IMDB tables. Some `nullable` declarations were not correct.

### CODE REVIEW

**Comment by Bouncner** [comment]
Do you know if performance has been impacted? With #2279, we removed some unnecessary NULL scans. Is that still the case? Or might we have even less IS NULL scans remain now?

**Comment by Bensk1** [comment]
> Do you know if performance has been impacted? With #2279, we removed some unnecessary NULL scans. Is that still the case? Or might we have even less IS NULL scans remain now?

No, I don't know but I can have a look at it.

**Comment by Bouncner** [comment]
If you have a diff with the changed lines, I can have a look if there are changes (have some table scan CSVs at hand).

**Comment by Bensk1** [comment]
A couple of attributes were changed to be not nullable anymore. PersonInfo's `note` field is now nullable.

<details>
<summary>Configuration Overview - click to expand</summary>

```diff
 +Configuration Overview---------------------------------------------+------------------------------------------------+
 | Parameter        | job_nullable_old.json                          | job_nullable_new.json                          |
 +------------------+------------------------------------------------+------------------------------------------------+
 |  GIT-HASH        | 54c2c122957f77ac0606ca2f031dd9d8ba76f9f8-dirty | e4bdb599a419976771071d08d9bc42aea2792c6f-dirty |
 |  benchmark_mode  | Ordered                                        | Ordered                                        |
 |  build_type      | release                                        | release                                        |
 |  chunk_size      | 65535                                          | 65535                                          |
 |  clients         | 1                                              | 1                                              |
 |  compiler        | gcc 9.3                                        | gcc 9.3                                        |
 |  cores           | 0                                              | 0                                              |
 |  date            | 2021-05-18 12:23:09                            | 2021-05-18 14:30:27                            |
 |  encoding        | {'default': {'encoding': 'Dictionary'}}        | {'default': {'encoding': 'Dictionary'}}        |
 |  indexes         | False                                          | False                                          |
 |  max_duration    | 60000000000                                    | 60000000000                                    |
 |  max_runs        | -1                                             | -1                                             |
 |  time_unit       | ns                                             | ns                                             |
 |  using_scheduler | False                                          | False                                          |
 |  verify          | False                                          | False                                          |
 |  warmup_duration | 0                                              | 0                                              |
 +------------------+------------------------------------------------+------------------------------------------------+
```
</details>

```diff
 +---------++----------+----------+--------++----------+----------+--------+---------+
 | Item    || Latency (ms/iter)   | Change || Throughput (iter/s) | Change | p-value |
 |         ||      old |      new |        ||      old |      new |        |         |
 +---------++----------+----------+--------++----------+----------+--------+---------+
 | 10a     ||    339.6 |    338.4 |   -0%  ||     2.94 |     2.96 |   +0%  |  0.7203 |
 | 10b     ||    328.6 |    329.0 |   +0%  ||     3.04 |     3.04 |   -0%  |  0.4323 |
 | 10c     ||   1011.1 |   1017.0 |   +1%  ||     0.99 |     0.98 |   -1%  |  0.0548 |
 | 11a     ||    186.3 |    187.0 |   +0%  ||     5.37 |     5.35 |   -0%  |  0.6725 |
 | 11b     ||    178.7 |    179.3 |   +0%  ||     5.59 |     5.58 |   -0%  |  0.7536 |
 | 11c     ||    599.0 |    597.8 |   -0%  ||     1.67 |     1.67 |   +0%  |  0.8377 |
 | 11d     ||   5607.6 |   5582.8 |   -0%  ||     0.18 |     0.18 |   +0%  |  0.7623 |
 | 12a     ||    313.7 |    327.5 |   +4%  ||     3.19 |     3.05 |   -4%  |  0.0009 |
 | 12b     ||    356.2 |    363.0 |   +2%  ||     2.81 |     2.76 |   -2%  |  0.1230 |
-| 12c     ||   3961.7 |   4164.8 |   +5%  ||     0.25 |     0.24 |   -5%  |  0.0004 |
 | 13a     ||    167.2 |    167.1 |   -0%  ||     5.98 |     5.99 |   +0%  |  0.5702 |
-| 13b     ||    366.7 |    439.9 |  +20%  ||     2.73 |     2.27 |  -17%  |  0.0000 |
-| 13c     ||    201.4 |    241.8 |  +20%  ||     4.97 |     4.14 |  -17%  |  0.0000 |
-| 13d     ||    150.7 |    167.0 |  +11%  ||     6.63 |     5.99 |  -10%  |  0.0000 |
 | 14a     ||   4524.8 |   4632.7 |   +2%  ||     0.22 |     0.22 |   -2%  |  0.0238 |
 | 14b     ||   5117.7 |   5222.9 |   +2%  ||     0.20 |     0.19 |   -2%  |  0.0488 |
 | 14c     ||   4507.3 |   4546.1 |   +1%  ||     0.22 |     0.22 |   -1%  |  0.7067 |
 | 15a     ||    210.1 |    207.7 |   -1%  ||     4.76 |     4.82 |   +1%  |  0.0411 |
 | 15b     ||    208.5 |    209.2 |   +0%  ||     4.80 |     4.78 |   -0%  |  0.3293 |
 | 15c     ||    204.7 |    204.7 |   +0%  ||     4.89 |     4.88 |   -0%  |  0.9954 |
 | 15d     ||    501.0 |    497.8 |   -1%  ||     2.00 |     2.01 |   +1%  |  0.1073 |
 | 16a     ||   4818.5 |   4908.4 |   +2%  ||     0.21 |     0.20 |   -2%  |  0.2742 |
 | 16b     ||   6646.3 |   6627.1 |   -0%  ||     0.15 |     0.15 |   +0%  |  0.8237 |
 | 16c     ||   5016.2 |   4994.8 |   -0%  ||     0.20 |     0.20 |   +0%  |  0.7743 |
 | 16d     ||   4954.5 |   5048.2 |   +2%  ||     0.20 |     0.20 |   -2%  |  0.3395 |
 | 17a     ||   1256.6 |   1255.7 |   -0%  ||     0.80 |     0.80 |   +0%  |  0.9079 |
 | 17b     ||    983.5 |    995.1 |   +1%  ||     1.02 |     1.00 |   -1%  |  0.0839 |
 | 17c     ||    931.4 |    944.9 |   +1%  ||     1.07 |     1.06 |   -1%  |  0.0279 |
 | 17d     ||   1138.5 |   1140.3 |   +0%  ||     0.88 |     0.88 |   -0%  |  0.8263 |
 | 17e     ||   3634.9 |   3575.4 |   -2%  ||     0.28 |     0.28 |   +2%  |  0.0929 |
 | 17f     ||   1901.0 |   1907.5 |   +0%  ||     0.53 |     0.52 |   -0%  |  0.6558 |
 | 18a     ||    649.3 |    651.5 |   +0%  ||     1.54 |     1.53 |   -0%  |  0.6487 |
 | 18b     ||    314.8 |    315.6 |   +0%  ||     3.18 |     3.17 |   -0%  |  0.6562 |
 | 18c     ||   4179.4 |   4363.6 |   +4%  ||     0.24 |     0.23 |   -4%  |  0.0000 |
 | 19a     ||   7307.2 |   7339.7 |   +0%  ||     0.14 |     0.14 |   -0%  |       ˅ |
 | 19b     ||   4212.2 |   4230.8 |   +0%  ||     0.24 |     0.24 |   -0%  |  0.1599 |
 | 19c     ||   7072.2 |   7107.0 |   +0%  ||     0.14 |     0.14 |   -0%  |       ˅ |
 | 19d     ||   5924.2 |   5958.5 |   +1%  ||     0.17 |     0.17 |   -1%  |  0.2570 |
 | 1a      ||    103.0 |    103.2 |   +0%  ||     9.71 |     9.69 |   -0%  |  0.0028 |
 | 1b      ||     23.1 |     23.3 |   +1%  ||    43.24 |    42.95 |   -1%  |  0.0000 |
 | 1c      ||     23.2 |     23.4 |   +1%  ||    43.04 |    42.82 |   -1%  |  0.0000 |
 | 1d      ||     23.1 |     23.4 |   +1%  ||    43.23 |    42.80 |   -1%  |  0.0000 |
 | 20a     ||   1736.3 |   1748.1 |   +1%  ||     0.58 |     0.57 |   -1%  |  0.0062 |
 | 20b     ||   1487.0 |   1498.0 |   +1%  ||     0.67 |     0.67 |   -1%  |  0.0004 |
 | 20c     ||   1046.6 |   1056.5 |   +1%  ||     0.96 |     0.95 |   -1%  |  0.0000 |
 | 21a     ||   4225.0 |   4357.2 |   +3%  ||     0.24 |     0.23 |   -3%  |  0.0000 |
 | 21b     ||    279.2 |    280.1 |   +0%  ||     3.58 |     3.57 |   -0%  |  0.2822 |
 | 21c     ||   4370.8 |   4510.2 |   +3%  ||     0.23 |     0.22 |   -3%  |  0.0000 |
-| 22a     ||   3755.1 |   3950.8 |   +5%  ||     0.27 |     0.25 |   -5%  |  0.0000 |
-| 22b     ||   3746.8 |   3946.6 |   +5%  ||     0.27 |     0.25 |   -5%  |  0.0000 |
 | 22c     ||   4508.2 |   4681.5 |   +4%  ||     0.22 |     0.21 |   -4%  |  0.0001 |
 | 22d     ||   4536.0 |   4683.7 |   +3%  ||     0.22 |     0.21 |   -3%  |  0.0000 |
 | 23a     ||    203.2 |    203.0 |   -0%  ||     4.92 |     4.93 |   +0%  |  0.7982 |
 | 23b     ||    111.7 |    114.9 |   +3%  ||     8.95 |     8.70 |   -3%  |  0.0000 |
 | 23c     ||    216.7 |    217.4 |   +0%  ||     4.61 |     4.60 |   -0%  |  0.4820 |
 | 24a     ||   4234.3 |   4241.7 |   +0%  ||     0.24 |     0.24 |   -0%  |  0.8028 |
 | 24b     ||   4138.5 |   4159.5 |   +1%  ||     0.24 |     0.24 |   -1%  |  0.7404 |
 | 25a     ||    312.2 |    314.7 |   +1%  ||     3.20 |     3.18 |   -1%  |  0.0507 |
 | 25b     ||    310.4 |    311.9 |   +0%  ||     3.22 |     3.21 |   -0%  |  0.1845 |
-| 25c     ||   4175.5 |   4374.8 |   +5%  ||     0.24 |     0.23 |   -5%  |  0.0000 |
 | 26a     ||    869.0 |    880.3 |   +1%  ||     1.15 |     1.14 |   -1%  |  0.0015 |
 | 26b     ||    855.9 |    869.1 |   +2%  ||     1.17 |     1.15 |   -2%  |  0.0245 |
 | 26c     ||    868.4 |    880.1 |   +1%  ||     1.15 |     1.14 |   -1%  |  0.0085 |
-| 27a     ||   3776.5 |   3972.7 |   +5%  ||     0.26 |     0.25 |   -5%  |  0.0000 |
-| 27b     ||   3739.2 |   3940.6 |   +5%  ||     0.27 |     0.25 |   -5%  |  0.0000 |
 | 27c     ||    222.3 |    223.1 |   +0%  ||     4.50 |     4.48 |   -0%  |  0.6337 |
 | 28a     ||   4555.0 |   4718.3 |   +4%  ||     0.22 |     0.21 |   -3%  |  0.1259 |
-| 28b     ||   3663.2 |   3874.6 |   +6%  ||     0.27 |     0.26 |   -5%  |  0.0453 |
 | 28c     ||   4573.3 |   4729.2 |   +3%  ||     0.22 |     0.21 |   -3%  |  0.1409 |
 | 29a     ||   4102.3 |   4121.0 |   +0%  ||     0.24 |     0.24 |   -0%  |  0.9326 |
 | 29b     ||    159.7 |    162.0 |   +1%  ||     6.26 |     6.17 |   -1%  |  0.7879 |
 | 29c     ||   4322.8 |   4344.2 |   +0%  ||     0.23 |     0.23 |   -0%  |  0.9373 |
 | 2a      ||    123.3 |    123.6 |   +0%  ||     8.11 |     8.09 |   -0%  |  0.0000 |
 | 2b      ||     99.5 |     99.8 |   +0%  ||    10.05 |    10.02 |   -0%  |  0.0019 |
 | 2c      ||     61.9 |     62.4 |   +1%  ||    16.16 |    16.03 |   -1%  |  0.0000 |
 | 2d      ||    159.1 |    159.1 |   +0%  ||     6.28 |     6.28 |   -0%  |  0.9041 |
 | 30a     ||    329.1 |    329.9 |   +0%  ||     3.04 |     3.03 |   -0%  |  0.8821 |
 | 30b     ||    799.6 |    802.7 |   +0%  ||     1.25 |     1.25 |   -0%  |  0.8359 |
-| 30c     ||   4187.4 |   4405.5 |   +5%  ||     0.24 |     0.23 |   -5%  |  0.0219 |
 | 31a     ||    346.6 |    347.0 |   +0%  ||     2.89 |     2.88 |   -0%  |  0.8914 |
 | 31b     ||    810.2 |    814.5 |   +1%  ||     1.23 |     1.23 |   -1%  |  0.6339 |
 | 31c     ||    350.2 |    350.1 |   -0%  ||     2.86 |     2.86 |   +0%  |  0.9771 |
 | 32a     ||     49.0 |     49.1 |   +0%  ||    20.40 |    20.35 |   -0%  |  0.0020 |
 | 32b     ||    336.4 |    336.0 |   -0%  ||     2.97 |     2.98 |   +0%  |  0.2858 |
 | 33a     ||     69.6 |     69.8 |   +0%  ||    14.36 |    14.33 |   -0%  |  0.7333 |
 | 33b     ||     68.7 |     68.9 |   +0%  ||    14.56 |    14.52 |   -0%  |  0.6391 |
 | 33c     ||     85.2 |     85.5 |   +0%  ||    11.73 |    11.70 |   -0%  |  0.6594 |
 | 3a      ||   4307.0 |   4479.6 |   +4%  ||     0.23 |     0.22 |   -4%  |  0.0000 |
+| 3b      ||     37.4 |     33.1 |  -11%  ||    26.76 |    30.21 |  +13%  |  0.0000 |
 | 3c      ||   5499.3 |   5546.9 |   +1%  ||     0.18 |     0.18 |   -1%  |  0.5069 |
 | 4a      ||    116.3 |    116.7 |   +0%  ||     8.60 |     8.57 |   -0%  |  0.4152 |
 | 4b      ||     25.9 |     26.1 |   +1%  ||    38.57 |    38.28 |   -1%  |  0.0000 |
 | 4c      ||    116.7 |    116.5 |   -0%  ||     8.57 |     8.58 |   +0%  |  0.5220 |
 | 5a      ||   4164.2 |   4320.8 |   +4%  ||     0.24 |     0.23 |   -4%  |  0.0000 |
 | 5b      ||    347.3 |    346.7 |   -0%  ||     2.88 |     2.88 |   +0%  |  0.3966 |
 | 5c      ||   4924.9 |   5078.7 |   +3%  ||     0.20 |     0.20 |   -3%  |  0.0000 |
 | 6a      ||    313.5 |    314.3 |   +0%  ||     3.19 |     3.18 |   -0%  |  0.0001 |
 | 6b      ||   1266.3 |   1271.4 |   +0%  ||     0.79 |     0.79 |   -0%  |  0.1330 |
 | 6c      ||    314.3 |    314.2 |   -0%  ||     3.18 |     3.18 |   +0%  |  0.6769 |
 | 6d      ||   1265.3 |   1269.6 |   +0%  ||     0.79 |     0.79 |   -0%  |  0.0745 |
 | 6e      ||    314.7 |    315.0 |   +0%  ||     3.18 |     3.17 |   -0%  |  0.5119 |
 | 6f      ||   1988.7 |   2033.2 |   +2%  ||     0.50 |     0.49 |   -2%  |  0.0000 |
 | 7a      ||    188.5 |    186.7 |   -1%  ||     5.31 |     5.36 |   +1%  |  0.4027 |
-| 7b      ||     96.7 |    107.1 |  +11%  ||    10.34 |     9.34 |  -10%  |  0.0000 |
-| 7c      ||  10516.4 |  11270.2 |   +7%  ||     0.10 |     0.09 |   -7%  |       ˅ |
 | 8a      ||    105.4 |    105.6 |   +0%  ||     9.48 |     9.47 |   -0%  |  0.6689 |
 | 8b      ||    227.2 |    227.2 |   +0%  ||     4.40 |     4.40 |   -0%  |  0.9389 |
 | 8c      ||   4649.9 |   4624.3 |   -1%  ||     0.22 |     0.22 |   +1%  |  0.4234 |
 | 8d      ||   1275.3 |   1271.3 |   -0%  ||     0.78 |     0.79 |   +0%  |  0.2683 |
 | 9a      ||   3553.2 |   3634.7 |   +2%  ||     0.28 |     0.28 |   -2%  |  0.0004 |
 | 9b      ||    410.8 |    412.3 |   +0%  ||     2.43 |     2.43 |   -0%  |  0.4838 |
 | 9c      ||   3570.6 |   3673.2 |   +3%  ||     0.28 |     0.27 |   -3%  |  0.0000 |
 | 9d      ||   4345.2 |   4412.8 |   +2%  ||     0.23 |     0.23 |   -2%  |  0.0075 |
 +---------++----------+----------+--------++----------+----------+--------+---------+
 | Sum     || 227073.5 | 232138.5 |   +2%  ||          |          |        |         |
 | Geomean ||          |          |        ||          |          |   -2%  |         |
 +---------++----------+----------+--------++----------+----------+--------+---------+
 |         || ˅ Insufficient number of runs for p-value calculation                  |
 +---------++----------+----------+--------++----------+----------+--------+---------+
```

**Comment by julianmenzler** [comment]
I wonder: What was our initial source for the table data and queries?

**Comment by Bensk1** [comment]
> I wonder: What was our initial source for the table data and queries?

I don't know exactly. @mrzzzrm did it and I believe it was based on the _official_ JOB repo.

**Comment by Bouncner** [comment]
Yes, if I am not mistaken we are completely following (except from things like `varchar(10)` the official repository). Moritz had even a commit on this repo to fix issues which caused problems in Hyrise's parser.

**Comment by Bouncner** [comment]
Which dropbox account is that?

File: scripts/setup_imdb.py [hunk 1]
```diff
@@ -35,7 +35,7 @@ def is_setup():
 assert len(sys.argv) == 2
 table_dir = sys.argv[1]
 
-LOCATION = "https://archive.org/download/imdb_20200624/imdb.zip"
+LOCATION = "http://www.dropbox.com/s/fgju7cifpehqlww/imdb.zip?dl=1"
 FILE_NAME = "imdb.zip"
```
Comment (julianmenzler) [inline scripts/setup_imdb.py:39]: This must be the original source then: https://github.com/gregrahn/join-order-benchmark/commits/master

I'd add this as a comment for reference.

**Comment by Bensk1** [comment]
> Yes, if I am not mistaken we are completely following (except from things like `varchar(10)` the official repository). Moritz had even a commit on this repo to fix issues which caused problems in Hyrise's parser.

The differences to the original repository in terms of nullability (we had nullable primary key columns in our version of the dataset) are the main motivation for this PR. Now, we should be completely following the official repo.

**Comment by Bensk1** [comment]
> Which dropbox account is that?

It's in the main folder of our research group's dropbox account and (on purpose) not bound to any specific employee. Dropbox is much faster and easier to update as the archive.org solution.

File: scripts/setup_imdb.py (hunk 1 above)
Comment (Bensk1) [reply scripts/setup_imdb.py:39]: Good idea.

**Review by Bouncner — APPROVED** [review]

**Review by dey4ss — APPROVED** [review]

**Pull request merged**